messaging/src/main/java/org/axonframework/eventhandling/gateway/ProcessingContextEventAppender.java

```diff
@@ -0,0 +1,81 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.gateway;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.configuration.NewConfiguration;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventSink;
+import org.axonframework.messaging.MessageTypeResolver;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Component that publishes events to an {@link EventSink} in the context of a {@link ProcessingContext}. The events
+ * will be published in the context this appender was created for. You can construct one through the
+ * {@link EventAppender#forContext(ProcessingContext, NewConfiguration)} method.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class ProcessingContextEventAppender implements EventAppender, DescribableComponent {
+
+    private final ProcessingContext processingContext;
+    private final EventSink eventSink;
+    private final MessageTypeResolver messageTypeResolver;
+
+    /**
+     * Creates a new {@link EventAppender} that uses the given {@code eventSink} to publish events. The
+     * {@code messageTypeResolver} is used to resolve the type of the event if no {@link EventMessage} is provided but a
+     * payload.
+     *
+     * @param processingContext   The {@link ProcessingContext} to publish events to.
+     * @param eventSink           The {@link EventSink} to publish events to.
+     * @param messageTypeResolver The {@link MessageTypeResolver} to resolve the type of the event.
+     */
+    ProcessingContextEventAppender(
+            ProcessingContext processingContext,
+            EventSink eventSink,
+            MessageTypeResolver messageTypeResolver
+    ) {
+        Objects.requireNonNull(eventSink, "EventSink may not be null");
+        Objects.requireNonNull(messageTypeResolver, "MessageTypeResolver may not be null");
+        this.processingContext = processingContext;
+        this.eventSink = eventSink;
+        this.messageTypeResolver = messageTypeResolver;
+    }
+
+    @Override
+    public void append(@Nonnull List<?> events) {
+        Objects.requireNonNull(events, "Events may not be null");
+        List<EventMessage<?>> eventMessages = events
+                .stream()
+                .map(e -> EventPublishingUtils.asEventMessage(e, messageTypeResolver))
+                .collect(Collectors.toList());
+        eventSink.publish(processingContext, eventMessages);
+    }
+
+    @Override
+    public void describeTo(@Nonnull ComponentDescriptor descriptor) {
+        descriptor.describeProperty("processingContext", processingContext);
```

Comment:
Just being certain here, but I think you deliberately didn't add the `EventSink` and `MessageTypeResolver` here, correct?

Reply:
Nope, I missed this. Adding.